# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: testing/web-platform/tests/intersection-observer/resources/nested-cross-origin-child-iframe.sub.html
# Commit: b8ad0267e88a
# Full Hash: b8ad0267e88abff0a54a6f0f1b5de2ad3a018460
# Author: Hiroyuki Ikezoe <hikezoe.birchill@mozilla.com>
# Date: 2020-02-11 15:57:30
# Regressor Bug: 1599795
# File Overlap Count: 1
# Description:
#   Bug 1599795 - Make IntersectionObserver work in fission world. r=emilio
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D61938
# ==============================================================================

diff -r 3985a60a75b3 -r b8ad0267e88a testing/web-platform/tests/intersection-observer/resources/nested-cross-origin-child-iframe.sub.html
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/testing/web-platform/tests/intersection-observer/resources/nested-cross-origin-child-iframe.sub.html	Tue Feb 11 03:59:14 2020 +0000
@@ -0,0 +1,22 @@
+<!DOCTYPE html>
+<script src="/common/get-host-info.sub.js"></script>
+<script src="/css/cssom-view/support/scroll-behavior.js"></script>
+<style>
+.spacer {
+  height: calc(100vh + 100px);
+}
+</style>
+<div class="spacer"></div>
+<iframe id="iframe"></iframe>
+<script>
+iframe.src =      // secure port
+  get_host_info().HTTPS_NOTSAMESITE_ORIGIN + "/intersection-observer/resources/nested-cross-origin-grand-child-iframe.html";
+
+window.addEventListener("message", async event => {
+  if (event.data == "scroll") {
+    iframe.scrollIntoView({ behavior: "instant" });
+    await waitForScrollEnd(document.scrollingElement);
+    window.parent.postMessage("scrollEnd", "*");
+  }
+});
+</script>